### PR TITLE
Massive Update \o/

### PR DESCRIPTION
**Pull request by Ghabry**
This refactors the build script and moves common stuff in "shared/" and "sources" package variables through a packages script.

Android: Removed MIPS, added arm64
Switch: Added, uses libsamplerate because speex didn't compile :(

Update stats:
Everything (except Windows) updated to latest library version.
Added Harfbuzz everywhere except emscripten (no freetype or harfbuzz)
Added Opus(file) everywhere except on 3DS (fast-math breaks it)
Except ICU which is 59.1 everywhere (Except Emscripten, 60.2) for now.

3ds and Vita: Added expat
Wii: Still at ICU 58.1 because of the complex big endian patches
iOS: SDl 2.0.7 doesn't compile, stay on 2.0.6

**Currently recompiling Player of all platforms to ensure it works**

### CODE REVIEW

**Comment by Ghabry** [comment]
All Players compile except Android (obviously!) which reports "undefined reference to 'atof'" for all 4 target platforms.

File: vita/2_build_toolchain.sh [hunk 1]
```diff
@@ -5,79 +5,72 @@ set -e
...
+	if [ "$ENABLE_CCACHE" ]; then
+		export CC="ccache $TARGET_HOST-gcc"
+		export CXX="ccache $TARGET_HOST-g++"
+	else
+		export CC="$TARGET_HOST-gcc"
+		export CXX="$TARGET_HOST-g++"
+	fi
+	export CFLAGS="-I$PLATFORM_PREFIX/include -g0 -O2 -DPSP2"
+	export CPPFLAGS=$CFLAGS
+	export CXXFLAGS=$CFLAGS
```
Comment (carstene1ns) [inline vita/2_build_toolchain.sh]: I know that has been in here before, but is really insane.
Only put `-I` and `-D` stuff in `CPPFLAGS`...

File: android/2_build_toolchain.sh [hunk 2]
```diff
@@ -5,108 +5,62 @@ set -e
...
-
 # Install mpg123
 function install_lib_mpg123 {
 	export CPPFLAGSOLD=$CPPFLAGS
 	export CPPFLAGS="$CPPFLAGS -DHAVE_MMAP"
-	install_lib mpg123-1.25.8 --enable-fifo=no --enable-ipv6=no --enable-network=no \
-		--enable-int-quality=no --with-cpu=generic --with-default-audio=dummy
+	rm $PLATFORM_PREFIX/config.cache
+	install_lib $MPG123_DIR $MPG123_ARGS
+	rm $PLATFORM_PREFIX/config.cache
```
Comment (carstene1ns) [inline android/2_build_toolchain.sh]: This totally breaks the purpose why we use a build system cache!
I would like to investigate why `mmap` is not detected/used instead.

File: android/4_build_liblcf.sh [hunk 3]
```diff
@@ -5,97 +5,94 @@ set -e
...
 else
 	git clone https://github.com/EasyRPG/liblcf.git
 fi
+
+# Number of CPU
+nproc=$(nproc)
+
+# Use ccache?
+test_ccache
+
```
Comment (carstene1ns) [inline android/4_build_liblcf.sh]: TL;DR: Too much copypasted content in here.

File: android/2_build_toolchain.sh (hunk 2 above)
Comment (Ghabry) [reply android/2_build_toolchain.sh]: good point, I will recompile to check if this was fixed in the meanwhile.

**Comment by carstene1ns** [comment]
Did you try adding `-lm` to the android makefile for `atof`?

**Comment by Ghabry** [comment]
-lm doesn't help. No idea what's going on as the NDK version used is the same as before.

File: shared/common [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+	shift
+	rm -rf CMakeCache.txt CMakeFiles/
+	$CMAKE_WRAPPER cmake . -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBUILD_SHARED_LIBS=OFF \
+		-DCMAKE_INSTALL_PREFIX=$PLATFORM_PREFIX $@
+	make clean
+	make
+	make install
+	popd
+
+	echo " -> done"
```
[the remark this answers is not in the document]
Comment (Ghabry) [reply shared/common]: because the linux build uses "Linux" as platform. But I can use Generic for all.

**Comment by Ghabry** [comment]
should be ready now

**Comment by Ghabry** [comment]
Because it's a security update I will make an exception and update libvorbis

**Comment by Ghabry** [comment]
also updated mpg123 because our patch applied without issues. Submitted that patch to upstream months ago but guess the dev forgot by now about it :/

**Pull request merged**